src/main/java/org/apache/mesos/hdfs/state/LiveState.java

```diff
@@ -1,103 +1,123 @@
+    Set<Protos.TaskInfo> toRemove = new HashSet<>();
+    for (Protos.TaskInfo taskInfo : stagingTasks ) {
+      if (taskInfo.getTaskId().equals(taskID)) {
+        toRemove.add(taskInfo);
+      }
+    }
+    stagingTasks.removeAll(toRemove);
+  }
+
+  public LinkedHashMap<Protos.TaskID, Protos.TaskStatus> getRunningTasks() {
+    return runningTasks;
   }
 
-  public Map<Protos.TaskID, String> getTaskSlaveMap() {
-    return taskSlaveMap;
+  public void removeTask(Protos.TaskID taskId) {
+    runningTasks.remove(taskId);
   }
 
-  public Set<Protos.TaskID> getJournalNodes() {
-    return journalNodes;
+  public void updateTaskForStatus(Protos.TaskStatus status) {
+    runningTasks.put(status.getTaskId(), status);
   }
 
-  public Set<Protos.TaskID> getNameNodes() {
-    return nameNodes;
+  public AcquisitionPhase getCurrentAcquisitionPhase() {
+    return currentAcquisitionPhase;
   }
 
-  public Set<String> getJournalNodeHosts() {
-    return journalNodeHosts;
+  public void transitionTo(AcquisitionPhase phase) {
+    this.currentAcquisitionPhase = phase;
   }
 
-  public Set<String> getNameNodeHosts() {
-    return nameNodeHosts;
+  public int getJournalNodeSize() {
+    return countOfRunningTasksWith(HDFSConstants.JOURNAL_NODE_ID);
   }
 
-  public Set<Protos.TaskID> getStagingTasks() {
-    return stagingTasks;
+  public int getNameNodeSize() {
+    return countOfRunningTasksWith(HDFSConstants.NAME_NODE_TASKID);
   }
 
-  public boolean notInDfsHosts(String slaveId) {
-    return !taskSlaveMap.values().contains(slaveId);
+  public Protos.TaskID getFirstNameNodeTaskId() {
+    if (getNameNodeSize() >= 1) {
+      return getNamenodeTaskIds().get(0);
+    } else {
+      return null;
+    }
+  }
+
+  public Protos.TaskID getSecondNameNodeTaskId() {
+    if (getNameNodeSize() == HDFSConstants.TOTAL_NAME_NODES) {
+      return getNamenodeTaskIds().get(1);
+    } else {
+      return null;
+    }
   }
 
-  public void addTask(Protos.TaskID taskId, String hostname, String slaveId) {
-    taskHostMap.put(taskId, hostname);
-    taskSlaveMap.put(taskId, slaveId);
-    if (taskId.getValue().contains(HDFSConstants.NAME_NODE_TASKID)) {
-      nameNodeHosts.add(hostname);
-    } else if (taskId.getValue().contains(HDFSConstants.JOURNAL_NODE_ID)) {
-      journalNodeHosts.add(hostname);
+  public Protos.SlaveID getFirstNameNodeSlaveId() {
+    if (getNameNodeSize() >= 1) {
+      return getNamenodeSlaveIds().get(0);
+    } else {
+      return null;
     }
   }
 
-  public void updateTask(Protos.TaskStatus taskStatus) {
-    if (taskStatus.getTaskId().getValue().contains(HDFSConstants.NAME_NODE_TASKID)) {
-      nameNodes.add(taskStatus.getTaskId());
-    } else if (taskStatus.getTaskId().getValue()
-        .contains(HDFSConstants.JOURNAL_NODE_ID)) {
-      journalNodes.add(taskStatus.getTaskId());
+  public Protos.SlaveID getSecondNameNodeSlaveId() {
+    if (getNameNodeSize() == HDFSConstants.TOTAL_NAME_NODES) {
```

Comment:
Ditto. I don't like this TOTAL_NAME_NODES constant unless we actually change the code everywhere to accept other values for it. Either hardcode it everywhere or nowhere. A hybrid approach is a recipe for disaster.